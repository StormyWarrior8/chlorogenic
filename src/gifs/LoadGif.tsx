import * as React from "react"
import styled from "../appearance/styled"

const Image = styled.img`
  width: 100%;
  box-shadow: 0 14px 28px rgba(0, 0, 0, 0.25), 0 10px 10px rgba(0, 0, 0, 0.22);
`

export type LoadGifProps = {
  gif: "add" | "move" | "sync"
} & React.DetailedHTMLProps<
  React.ImgHTMLAttributes<HTMLImageElement>,
  HTMLImageElement
>

export default class extends React.PureComponent<LoadGifProps, { file?: any }> {
  public state = { file: undefined }
  public componentDidMount() {
    import(`./${this.props.gif}.gif`)
      .then(file => this.setState({ file }))
      .catch(error => {
        // tslint:disable-next-line:no-console
        console.error(error)
      })
  }

  public render() {
    return <Image src={this.state.file} alt="Loading..." />
  }
}
