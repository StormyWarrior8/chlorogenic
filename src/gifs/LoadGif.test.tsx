import * as React from "react"
import { render, wait } from "react-testing-library"
import LoadGif, { LoadGifProps } from "./LoadGif"

it("snapshot", async () => {
  const gifs: Array<LoadGifProps["gif"]> = ["add", "move", "sync"]
  for (const gif of gifs) {
    const { container, queryByAltText } = render(<LoadGif gif={gif} />)

    expect(container.firstChild).toMatchSnapshot()

    await wait(() => {
      const img: HTMLImageElement = queryByAltText(
        "Loading...",
      ) as HTMLImageElement
      return expect(img.src).not.toBeFalsy()
    })

    expect(container.firstChild).toMatchSnapshot()
  }
})
