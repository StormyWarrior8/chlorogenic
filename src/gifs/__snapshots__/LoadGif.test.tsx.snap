// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<img
  alt="Loading..."
  class="sc-bdVaJa cuYrOR"
/>
`;

exports[`snapshot 2`] = `
<img
  alt="Loading..."
  class="sc-bdVaJa cuYrOR"
  src="add.gif"
/>
`;

exports[`snapshot 3`] = `
<img
  alt="Loading..."
  class="sc-bdVaJa cuYrOR"
/>
`;

exports[`snapshot 4`] = `
<img
  alt="Loading..."
  class="sc-bdVaJa cuYrOR"
  src="move.gif"
/>
`;

exports[`snapshot 5`] = `
<img
  alt="Loading..."
  class="sc-bdVaJa cuYrOR"
/>
`;

exports[`snapshot 6`] = `
<img
  alt="Loading..."
  class="sc-bdVaJa cuYrOR"
  src="sync.gif"
/>
`;
